airflow/providers/google/cloud/operators/cloud_composer.py

```diff
@@ -0,0 +1,696 @@
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :param retry: Designation of what errors, if any, should be retried.
+    :param timeout: The timeout for this request.
+    :param metadata: Strings which should be sent along with the request as metadata.
+    :param deferrable: Run operator in the deferrable mode
+    :param pooling_period_seconds: Optional: Control the rate of the poll for the result of deferrable run.
+        By default the trigger will poll every 30 seconds.
+    """
+
+    template_fields = (
+        'project_id',
+        'region',
+        'environment_id',
+        'environment',
+        'impersonation_chain',
+    )
+
+    operator_extra_links = (CloudComposerEnvironmentLink(),)
+
+    def __init__(
+        self,
+        *,
+        project_id: str,
+        region: str,
+        environment_id: str,
+        environment: Union[Environment, Dict],
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        delegate_to: Optional[str] = None,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Sequence[Tuple[str, str]] = (),
+        deferrable: bool = False,
+        pooling_period_seconds: int = 30,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.project_id = project_id
+        self.region = region
+        self.environment_id = environment_id
+        self.environment = environment
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+        self.delegate_to = delegate_to
+        self.deferrable = deferrable
+        self.pooling_period_seconds = pooling_period_seconds
+
+    def execute(self, context: 'Context'):
+        hook = CloudComposerHook(
+            gcp_conn_id=self.gcp_conn_id,
+            impersonation_chain=self.impersonation_chain,
+            delegate_to=self.delegate_to,
+        )
+
+        name = hook.get_environment_name(self.project_id, self.region, self.environment_id)
+        if isinstance(self.environment, Environment):
+            self.environment.name = name
+        else:
+            self.environment["name"] = name
+
+        self.xcom_push(
+            context,
+            key='composer_conf',
+            value={
+                "project_id": self.project_id,
+                "region": self.region,
+                "environment_id": self.environment_id,
+            },
+        )
```

Comment:
Same comment as here https://github.com/apache/airflow/pull/21267#discussion_r800086099, the minimal change I would expect is to extract the key value as class link attribute so we cane reference it instead of hardcoding it in two places.

Reply:
@turbaszek Good idea, already changed.